core/src/main/java/org/geysermc/geyser/entity/type/living/ArmorStandEntity.java

```diff
@@ -362,7 +362,7 @@ private void updateSecondEntityStatus(boolean sendMetadata) {
             setFlag(EntityFlag.INVISIBLE, false);
             dirtyMetadata.put(EntityData.SCALE, 0.0f);
             // As the above is applied, we need an offset
-            updateOffsetRequirement(true);
+            updateOffsetRequirement(false);
```

Comment:
This seems to fix the passenger's position, however it causes the name tag of the armor stand to appear far too low.

Reply:
the reason this was changed to false was because it was changing the offset of the armorstand when it was set to invisible

Reply:
Since this offset is required to position the name tags for invisible armor stands correctly, you should instead counteract this offset for passengers within updateMountOffset.